packages/js/src/plugins/auctionHouseModule/cancelBid.ts

```diff
@@ -0,0 +1,152 @@
+import { ConfirmOptions, SYSVAR_INSTRUCTIONS_PUBKEY } from '@solana/web3.js';
+import type { Metaplex } from '@/Metaplex';
+import { TransactionBuilder } from '@/utils';
+import {
+  CancelInstructionAccounts,
+  createCancelBidReceiptInstruction,
+  createCancelInstruction,
+  createAuctioneerCancelInstruction,
+} from '@metaplex-foundation/mpl-auction-house';
+import {
+  useOperation,
+  Operation,
+  OperationHandler,
+  Signer,
+  isSigner,
+  toPublicKey,
+  Pda,
+} from '@/types';
+import { SendAndConfirmTransactionResponse } from '../rpcModule';
+import { AuctionHouse } from './AuctionHouse';
+import { Bid } from './Bid';
+import { AuctioneerAuthorityRequiredError } from './errors';
+import { findAssociatedTokenAccountPda } from '../tokenModule';
+import { isNftWithToken, isSftWithToken } from '../nftModule';
+import { findAuctioneerPda } from './pdas';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'CancelBidOperation' as const;
+export const cancelBidOperation = useOperation<CancelBidOperation>(Key);
+export type CancelBidOperation = Operation<
+  typeof Key,
+  CancelBidInput,
+  CancelBidOutput
+>;
+
+export type CancelBidInput = {
+  auctionHouse: AuctionHouse;
+  auctioneerAuthority?: Signer; // Use Auctioneer ix when provided
+  bid: Bid;
+
+  // Options.
+  confirmOptions?: ConfirmOptions;
+};
+
+export type CancelBidOutput = {
+  response: SendAndConfirmTransactionResponse;
+};
+
+// -----------------
+// Handler
+// -----------------
+
+export const cancelBidOperationHandler: OperationHandler<CancelBidOperation> = {
+  handle: async (operation: CancelBidOperation, metaplex: Metaplex) =>
+    cancelBidBuilder(operation.input).sendAndConfirm(
+      metaplex,
+      operation.input.confirmOptions
+    ),
+};
+
+// -----------------
+// Builder
+// -----------------
+
+export type CancelBidBuilderParams = Omit<CancelBidInput, 'confirmOptions'> & {
+  instructionKey?: string;
+};
+
+export type CancelBidBuilderContext = Omit<CancelBidOutput, 'response'>;
+
+export const cancelBidBuilder = (
+  params: CancelBidBuilderParams
+): TransactionBuilder<CancelBidBuilderContext> => {
+  const { auctionHouse, auctioneerAuthority, bid } = params;
+
+  if (auctionHouse.hasAuctioneer && !auctioneerAuthority) {
+    throw new AuctioneerAuthorityRequiredError();
+  }
+
+  // Data.
+  const { asset, tradeStateAddress, price, tokens } = bid;
+
+  // Accounts.
+  const tokenAccount =
+    isNftWithToken(asset) || isSftWithToken(asset)
```

Comment:
We've got a boolean for that check.

```suggestion
    !asset.isPublic
```